assets/admin/students/student-modal/index.js

```diff
@@ -15,74 +15,96 @@ import httpClient from '../../lib/http-client';
+			sendAction: ( students, courses ) =>
+				httpClient( {
+					restRoute: '/sensei-internal/v1/course-students/batch',
+					method: 'POST',
+					data: { student_ids: students, course_ids: courses },
+				} ),
+			isDestructive: false,
+		},
+		remove: {
+			// Translators: placeholder is the number of selected students for plural, student's name for singular.
+			description: _n(
+				'Select the course(s) you would like to remove <strong>%1$s</strong> from:',
+				'Select the course(s) you would like to remove <strong>%1$d students</strong> from:',
+				studentCount,
+				'sensei-lms'
+			),
+			buttonLabel: __( 'Remove from Course', 'sensei-lms' ),
+			sendAction: ( students, courses ) =>
+				httpClient( {
+					restRoute: '/sensei-internal/v1/course-students/batch',
+					method: 'DELETE',
+					data: { student_ids: students, course_ids: courses },
+				} ),
+			isDestructive: true,
+		},
+		'reset-progress': {
+			// Translators: placeholder is the number of selected students for plural, student's name for singular.
+			description: _n(
+				'Select the course(s) you would like to reset or remove progress from for <strong>%1$s</strong>:',
+				'Select the course(s) you would like to reset or remove progress from for <strong>%1$d students</strong>:',
+				studentCount,
+				'sensei-lms'
+			),
+			buttonLabel: __( 'Reset or Remove Progress', 'sensei-lms' ),
+			sendAction: ( students, courses ) =>
+				httpClient( {
+					restRoute: '/sensei-internal/v1/course-progress/batch',
+					method: 'DELETE',
+					data: { student_ids: students, course_ids: courses },
+				} ),
 
-		isDestructive: true,
-	},
+			isDestructive: true,
+		},
+	};
+	return possibleActions[ action ];
 };
 
 /**
  * Questions modal content.
  *
  * @param {Object}   props
- * @param {Object}   props.action   Action that is being performed.
- * @param {Function} props.onClose  Close callback.
- * @param {Array}    props.students A list of Student ids related to the action should be applied.
+ * @param {Object}   props.action      Action that is being performed.
+ * @param {Function} props.onClose     Close callback.
+ * @param {Array}    props.students    A list of Student ids related to the action should be applied.
+ * @param {string}   props.studentName Name of the student, shown when there's only one student.
  */
-
-export const StudentModal = ( { action, onClose, students } ) => {
-	const {
-		description,
-		buttonLabel,
-		sendAction,
-		isDestructive,
-	} = POSSIBLE_ACTIONS[ action ];
+export const StudentModal = ( { action, onClose, students, studentName } ) => {
+	const { description, buttonLabel, sendAction, isDestructive } = getAction(
+		action,
+		students.length
+	);
 	const [ selectedCourses, setCourses ] = useState( [] );
 	const [ isSending, setIsSending ] = useState( false );
 	const [ hasError, setError ] = useState( false );
 	const isMounted = useRef( true );
+	const replacedDescription = useMemo( () => {
+		const replacementString =
```

Comment:
Do we need useMemo, I wouldn't say this is the most expensive calculation and it makes initial rendering slower. Apparently 19%

Reply:
Knew this could cause some performance issue, but this much! :P Thanks! Can you kindly take a look again? I've updated the PR